### ISSUE TITLE
PieChart method

### ISSUE THREAD

**Issue by OneFreemanWill** [EvotecIT/PSWriteOffice#19]
Hello Przemysław,

I have been struggling to use `.AddPieChart` and `.AddChartPie` to add a pie chart to Word doc output.

`.AddPieChart` seems to work OK in that my output doc lands up with a blank section that would be a pie, but whenever I try to use `.AddChartPie` I get an overload error.
 
 Most basic example:
```powershell
$PieChart = $DocumentTest.AddPieChart("Test Pie Chart", $true, 600, 300)

$PieChart.AddChartPie("Category1", 10)
$PieChart.AddChartPie("Category2", 20)
$PieChart.AddChartPie("Category3", 30)
```
Error:
```text
Error: Cannot find an overload for "AddChartPie" and the argument count: "2".
```

I've tried making sure I'm using the expected types as defined below, but every variation seems to fail and results in the overload error.
```powershell
[void] AddChartPie[T](
    [string] $name,
    [int[]] $values)

[void] AddChartPie[T](
    [string] $name,
    [List[T]] $values)
```
Can you offer any guidance or share an example?

**Comment by PrzemyslawKlys** [comment]
Ok, i checked it and it seems the implementation differs a bit from my other projects. 

The trick is - it's one name, but with 4 categories, and the values match the category... 

```cs
                List<string> categories = new List<string>() { "Food", "Housing", "Mix", "Data" };
                var pieChart2 = document.AddPieChart();
                pieChart2.AddCategories(categories);
                pieChart2.AddChartPie("Poland", new List<int> { 10, 20, 30, 40 });
```

I need to decide what to do, as the implementation is like that in all chartts - need to think about it, but till then - using categories with one entry should work.

![image](https://github.com/EvotecIT/PSWriteOffice/assets/15063294/0ec4be28-866c-41b0-978e-b521cb342963)

**Comment by PrzemyslawKlys** [comment]
This PR should resolve your issue:
- https://github.com/EvotecIT/OfficeIMO/pull/223

It also changes how things work when it comes to charts so you will  need to change some code, and then you need to null out values as it's possible to do AddPie(5).AddPie(7).AddPie(9) but this means an object is returned on AddPie, AddLine and so on.

**Comment by PrzemyslawKlys** [comment]
Yes, the whole charts have been rebuilt. I need to further improve it, but upgrading PSWriteOffice to new version will break current charts.

**Comment by PrzemyslawKlys** [comment]
This is now exposed through `Add-OfficeWordChart`, including pie charts.

```powershell
$rows = @(
    [pscustomobject]@{ Category = 'Compliant'; Count = 42 }
    [pscustomobject]@{ Category = 'Non-compliant'; Count = 8 }
)

New-OfficeWord -Path .\pie.docx {
    Add-OfficeWordChart -Type Pie -Data $rows -CategoryProperty Category -SeriesProperty Count -Title 'Compliance'
}
```

Breaking-change note: the old direct OfficeIMO chart pattern around `AddPieChart` / `AddChartPie` was rebuilt. Existing scripts using those low-level methods may need to move to `Add-OfficeWordChart` or the newer OfficeIMO chart API. Closing as implemented.
